src/main/com/algos/src/datastructures/caches/LRUCache.java

```diff
@@ -32,27 +21,27 @@ public LRUCache(int cap) {
 
     private void setCapacity(int newCapacity) {
         checkCapacity(newCapacity);
-        for (int i = data.size(); i > newCapacity; i--) {
-            Entry<K, V> evicted = evict();
-            data.remove(evicted.getKey());
-        }
         this.cap = newCapacity;
+        
     }
 
     private Entry<K, V> evict() {
         if (head == null) {
-            throw new RuntimeException("cache cannot be empty!");
+            throw new RuntimeException("Cache cannot be empty!");
         }
         Entry<K, V> evicted = head;
         head = evicted.getNextEntry();
-        head.setPreEntry(null);
+        if (head != null) {
+            head.setPreEntry(null);
+        }
         evicted.setNextEntry(null);
+        
         return evicted;
     }
 
     private void checkCapacity(int capacity) {
         if (capacity <= 0) {
-            throw new RuntimeException("capacity must greater than 0!");
+            throw new RuntimeException("Capacity must be greater than 0!");
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/rishabhbohra3/Algo-tutorial-test/pull/6#issuecomment-2211218148>#c206b8</a><span> - 07/05/2024, 05:54 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The use of 'RuntimeException' for capacity validation is not specific enough. It is better to use a more specific exception type to clearly indicate the nature of the error.
</div>
<div id="fix">
<b>Fix:</b> Replace 'RuntimeException' with 'IllegalArgumentException' to provide a more specific exception type for invalid capacity.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -42,7 +42,7 @@
  private void checkCapacity(int capacity) {
      if (capacity <= 0) {
 -        throw new RuntimeException("Capacity must be greater than 0!");
 +        throw new IllegalArgumentException("Capacity must be greater than 0!");
      }
  }
```
</div>
</details>
</div>
<br>


</div>